docs/deployer/terraform.md

```diff
@@ -0,0 +1,168 @@
+# Terraform Deployer
+
+The Terraform deployer is a controller that reconciles
+DeployItems of type `landscaper.gardener.cloud/terraform`.
+
+It manages infrastructure resources using terraform.
+This means running the `apply` and `destroy` terraform commands.
+
+To run these commands, it relies on the Gardener component
+[Terraformer](https://github.com/gardener/terraformer)
+
+### Configuration
+
+This sections describes the provider specific configuration
+
+```yaml
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: DeployItem
+metadata:
+  name: s3-bucket
+spec:
+  type: landscaper.gardener.cloud/terraform
+
+  target: # has to be of type landscaper.gardener.cloud/kubernetes-cluster
+    name: my-cluster
+    namespace: test
+
+  config:
+    apiVersion: terraform.deployer.landscaper.gardener.cloud/v1alpha1
+    kind: ProviderConfiguration
+
+    # Use a specific image for the Terraformer
+    # optional
+    terraformerImage: eu.gcr.io/gardener-project/gardener/terraformer:v1.5.0
+
+    # base64 encoded kubeconfig pointing to the cluster to install the chart
+    # optional if Target is defined.
+    kubeconfig: <redacted>
```

Comment:
I see, the way it's done here, it allows the terraformer to run on a different cluster where the secrets are and that's also where it will store its configuration within the namespace. if it has to run on the cluster and in the name the deployitem is deployed, it makes the thing even easier.

Reply:
> If it has to run on the cluster and in the namespace where the DeployItem is deployed, it makes the thing even easier.

Its not the cluster where the deployitem lives. That could be another cluster.
It should be cluster where the controller is deployed. (This can be same cluster but this is not a must)